agent/service.go

```diff
@@ -204,25 +207,41 @@ func (as *agentService) Attestation(ctx context.Context) ([]byte, error) {
 }
 
 func (as *agentService) runComputation() {
-	ctx := context.Background()
-	as.publishEvent(ctx, "running", "computation run has started")()
+	as.publishEvent("starting", json.RawMessage{})()
 	as.sm.logger.Debug("computation run started")
 	defer as.sm.SendEvent(runComplete)
+	ctx := context.Background()
 	var cancel context.CancelFunc
 	if as.computation.Timeout.Duration != 0 {
 		ctx, cancel = context.WithDeadline(ctx, time.Now().Add(as.computation.Timeout.Duration))
 		defer cancel()
 	}
+	as.publishEvent("in-progress", json.RawMessage{})()
 	result, err := run(ctx, as.algorithms[0], as.datasets[0])
 	if err != nil {
 		as.runError = err
+		detail := struct {
```

Comment:
What is this that is called `detail`? It is not a good and self-describing name.

Reply:
Details field is an instructed field allowing to add event details as raw json